libcontainer/nsenter/nsexec.c

```diff
@@ -912,6 +1119,26 @@ void nsexec(void)
 			if (unshare(config.cloneflags & ~CLONE_NEWCGROUP) < 0)
 				bail("failed to unshare namespaces");
 
+			/* Ask our parent to send the mount sources fds. */
```

Comment:
> Could it make sense to add a conditional here to ask the parent only when it's really required?

Ok but I think the test should be on `config->mountsources == NULL`, which is set by libcontainer/container_linux.go. Otherwise, the boolean logic to know whether it's required would be duplicated in libcontainer/container_linux.go and libcontainer/nsenter/nsexec.c and it will be more difficult to keep in sync.

> if ((config.cloneflags & CLONE_NEWUSER) && (config.cloneflags & CLONE_NEWNS)) ?

It also depends on `c.config.RootlessEUID` and `len(c.config.Mounts) == 0`, it reinforces the previous point.

Reply:
Oh yes, that makes sense.